kotlinx-collections-immutable/src/main/kotlin/kotlinx/collections/immutable/implementations/immutableMap/PersistentHashMapBuilder.kt

```diff
@@ -24,7 +24,13 @@ internal class PersistentHashMapBuilder<K, V>(private var map: PersistentHashMap
     internal var marker = Marker()
     internal var node = map.node
     internal var operationResult: V? = null
+    internal var modCount = 0
+
     override var size = map.size
+        set(value) {
+            field = value
+            modCount++
+        }
```

Comment:
Is it possible that operations that do not modify size (e.g. `Map.put`) nevertheless invalidate internal structure of the map, so its iterators are becoming invalid after that?

Reply:
It turns out it's possible. Only in case of `Map.put` and `List.set`, so I fixed them.